src/main/java/org/candlepin/insights/inventory/InventoryService.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (c) 2009 - 2016 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.insights.inventory.client;
+
+import org.candlepin.insights.exceptions.RhsmConduitException;
+import org.candlepin.insights.inventory.client.model.FactSet;
+import org.candlepin.insights.inventory.client.model.Host;
+import org.candlepin.insights.inventory.client.model.HostOut;
+import org.candlepin.insights.inventory.client.resources.HostsApi;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * A wrapper for the insights inventory client.
+ */
+@Component
+public class InventoryService {
+
+    private static Logger log = LoggerFactory.getLogger(InventoryService.class);
+
+    private HostsApi hostsInventoryApi;
+
+    @Autowired
+    public InventoryService(ApiClientFactory clientFactory) throws Exception {
```

Comment:
Instead of wiring in the factory, it's more idiomatic to wire in a client (i.e. let Spring call `clientFactory.getObject()` for you).

Reply:
DONE - Will be included in next commit.